src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs

```diff
@@ -454,6 +454,62 @@ public Task AllowPreflightRequest()
             RedirectToLeader();
         }
 
+        private static void ValidateUrl(string nodeUrl, ClusterTopology clusterTopology)
+        {
+            var possibleNode = clusterTopology.TryGetNodeTagByUrl(nodeUrl);
+
+            if (possibleNode.HasUrl)
+            {
+                throw new InvalidOperationException($"Can't add a new node on {nodeUrl} to cluster because this url is already used by node {possibleNode.NodeTag}");
+            }
+
+            var uri = new Uri(nodeUrl);
+            var host = uri.DnsSafeHost;
+
+            var isHttps = nodeUrl.StartsWith("https://", StringComparison.InvariantCultureIgnoreCase);
+
+            if (IPAddress.TryParse(host, out IPAddress ipAddress))
+            {
+                possibleNode = clusterTopology.TryGetNodeTagByUrl(GetFullAddress(isHttps, ipAddress, uri.Port));
+                if (possibleNode.HasUrl)
+                {
+                    throw new InvalidOperationException($"Can't add a new node on {nodeUrl} to cluster because this url is already used by node {possibleNode.NodeTag}");
+                }                
+            }            
+            
+            IPHostEntry ipHostEntry = null;
+            try
+            {
+                ipHostEntry = Dns.GetHostEntry(host);
```

Comment:
No, there are too many ways in which this can go wrong.
The way to validate that we aren't adding the same node twice is to _go ask that node_ and see if it is there already